src/InventoryManager.cpp

```diff
@@ -194,44 +194,53 @@ int InventoryManager::userInput() {
 
         valid_transaction = false;
 
-        std::cout << "Buyer | Seller\n";
-        std::cin >> buyer >> seller;
-        std::cout << "Enter Q for Item name or 0 for Quantity Sold to stop reading sales in the transaction\n";
+        std::cout << "\nCustomer name: ";
+        std::cin >> buyer;
+        std::cout << "Seller name (you): ";
+        std::cin >> seller;
 
-        sale_list->userTransaction(sale_list->curr_sale_id, buyer, seller);
+        do {
+            sale_list->userTransaction(sale_list->curr_sale_id, buyer, seller);
 
-        while (true) {
-            std::cout << "Item Name: ";
+            std::cout << "\nItem Name: ";
             std::cin >> name;
-
-            if (name == "Q" || name == "q") break;
-
+ 
             std::cout << "Quantity Sold: ";
             std::cin >> quantity;
 
```

Comment:
You still need to check for quantity being greater than 0. Otherwise we will be adding sales with 0 items sold.
You could do something like :
int check = stoi(quantity);
if (check < 0) {
  std::cout << "Quantity cannot be  negative.\n";
} else if (check == 0) break;

Or something along those lines.